src/main/java/com/klodnicki/Bike/v2/service/BikeService.java

```diff
@@ -78,7 +66,7 @@ public void deleteById(Long id) {
     @Override
     public BikeForAdminResponseDTO update(Long id, BikeRequestDTO updatedBikeRequestDTO) {
 
-        Bike bike = bikeRepository.findById(id).orElseThrow(IllegalArgumentException::new);
+        Bike bike = modelMapper.map(findById(id), Bike.class);
```

Comment:
This is rather smelly, as you're now double mapping it in two ways due to the fact that your findById method doesn't return Bike but rather "BikeForAdminResponseDTO" which is causing you this issue.
It'd be much nicer if method used here would return a Bike so conversion would not be needed. It's just wasting time to map once just to map to the same Bike as before. Just looking at this should bring you the idea that it shouldn't be like that.
Please, keep in mind things like that in the future.

Reply:
Ok then I just extract a method for finding a bike in repo. Check it out now!